react-modal/my-app/src/App.tsx

```diff
@@ -0,0 +1,30 @@
+import './App.css';
+import { useRef } from 'react';
+
+function App() {
+  const modal = useRef<HTMLDialogElement>(null);
+
+  function handleOpenClick() {
+    modal.current?.showModal();
+  }
+  function handleClose() {
+    modal.current?.close();
+  }
+  function handleDelete() {
+    window.alert();
+    modal.current?.close();
+  }
+
+  return (
+    <>
+      <button onClick={handleOpenClick}>Delete Me!</button>
+      <dialog ref={modal}>
+        Do you really want to delete?
+        <button onClick={handleClose}>Cancel</button>
+        <button onClick={handleDelete}>Delete</button>
+      </dialog>
+    </>
+  );
+}
+
```

Comment:
It would be better to manage the open/close behavior of the modal using state. Since whether the modal is open directly affects how it's rendered, using a boolean state variable (e.g. `isModalOpen`) makes the component more declarative and easier to reason about. We can then pair `useState` with `useEffect` to trigger those methods when the state changes. That way, the visibility is driven by React state rather than direct DOM manipulation.